crates/blockchain/mempool.rs

```diff
@@ -249,23 +249,25 @@ impl Mempool {
         nonce: u64,
         received_hash: H256,
     ) -> Result<Option<MempoolTransaction>, MempoolError> {
-        let pooled_transactions = self
+        let Some(hash) = self
+            .txs_by_sender_nonce
+            .read()
+            .map_err(|error| StoreError::MempoolReadLock(error.to_string()))?
+            .get(&(sender, nonce))
+            .cloned()
+        else {
+            return Ok(None);
+        };
+        if hash == received_hash {
+            return Ok(None);
+        }
```

Comment:
I'm not sure this reflects the behaviour we want out of the mempool or the function, but I understand that for this PR we're trying to not modify how things worked before, just make them faster.